### PR TITLE
[FR] Selectionmarks

### LINKED ISSUE #14816: Support selection marks
- [x] As type on FormField
- [x] As a list[SelectionMarks] on FormPage
- [x] As an element type on FieldData/FormTableCell - requires parsing the json pointer
- [x] tests
- [x] samples
- [x] documentation
- [x] mocking

### PR DESCRIPTION
**Pull request by maririos**
Includes additions to: 
- As type in FormField
- As a list on FormPage
- As an element type on FieldData/FormTableCell - requires parsing the json pointer
- Mocking
- Tests
- Samples
- Documentation

Fixes: https://github.com/Azure/azure-sdk-for-net/issues/14816

~~Making it draft until I get green light to share the files for training/test selection mark~~
Test are disabled until we can upload the file here. I created issue https://github.com/Azure/azure-sdk-for-net/issues/15495 to track that

### CODE REVIEW

File: sdk/formrecognizer/Azure.AI.FormRecognizer/README.md [hunk 1]
```diff
@@ -160,6 +160,17 @@ foreach (FormPage page in formPages)
             Console.WriteLine($"    Cell ({cell.RowIndex}, {cell.ColumnIndex}) contains text: '{cell.Text}'.");
         }
     }
+
```
Comment (samvaity) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/README.md:163]: Do we want to show an example of `asSelectionMarkState` usage in one of the samples?
Sorry if it's already there and I have missed it..

Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/README.md:163]: What is `asSelectionMarkState` ? not sure I see the equivalent in .NET

Comment (samvaity) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/README.md:163]: here - https://github.com/Azure/azure-sdk-for-net/pull/15248/files#diff-8e44b539efa928d1579e9e5d814789c9R375

Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/README.md:163]: Ahh I see. So the only sample we currently showcase the other `asXXtype` is strongly typed `Receipt` sample. 
I don't see a receipt normally having selection marks. it is usually output data.
For now I will say no.
If customers ask for samples on how to use it then we can consider it

File: sdk/formrecognizer/Azure.AI.FormRecognizer/README.md [hunk 2]
```diff
@@ -137,7 +137,7 @@ The following section provides several code snippets illustrating common pattern
 * [Manage Custom Models Synchronously](#manage-custom-models-synchronously)
 
 ### Recognize Content
-Recognize text and table data, along with their bounding box coordinates, from documents.
+Recognize text, table, and and selection marks like radio buttons and check boxes data, along with their bounding box coordinates, from documents.
```
Comment (kinelski) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/README.md:140]: Since `selection markS` is pluralized, it would make more sense to say `texts, tables, ...` (or make everything singular).
The `like radio buttons and check boxes` makes it hard to parse the sentence.

`...text, table and selection mark data` sounds better to me, but I understand you're trying to be clear.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormSelectionMark.cs [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+namespace Azure.AI.FormRecognizer.Models
+{
+    /// <summary>
+    /// Represents a Selection mark recognized from the input document.
+    /// </summary>
+    public class FormSelectionMark : FormElement
+    {
+        internal FormSelectionMark(SelectionMark selectionMark, int pageNumber)
+            : base(new FieldBoundingBox(selectionMark.BoundingBox), pageNumber, selectionMark.State.ToString())
```
[the remark this answers is not in the document]
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormSelectionMark.cs:12]: Yes. we are all doing it.
The service only returns information about the checkbox, not the text attached to it. although because this is only for labeled models, the user can access that text too.
If we leave the property empty then we wouldn't be able to access the value of the selection mark when [looking](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/formrecognizer/Azure.AI.FormRecognizer/samples/Sample2_RecognizeCustomForms.md#recognize-custom-forms-from-a-uri) at the `RecognizedForm.Fields.ValueData` i.e. `Console.WriteLine($"    Value: '{field.ValueData.Text}");`

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeContentOperation.cs [hunk 4]
```diff
@@ -181,7 +181,7 @@ private async ValueTask<Response> UpdateStatusAsync(bool async, CancellationToke
 
                     if (update.Value.Status == OperationStatus.Succeeded)
                     {
-                        // we need to first assign a vaue and then mark the operation as completed to avoid race conditions
+                        // we need to first assign a value and then mark the operation as completed to avoid race conditions
```
[the remark this answers is not in the document]
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeContentOperation.cs:184]: yes :( :)

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs [hunk 5]
```diff
@@ -812,6 +862,43 @@ public async Task StartRecognizeCustomFormsWithLabels(bool useStream, bool inclu
...
+                expectedFirstPageNumber: 1, expectedLastPageNumber: 1);
+
+            // Testing that we shuffle things around correctly so checking only once per property.
+            Assert.IsNotEmpty(form.FormType);
+            Assert.AreEqual(1, form.Pages.Count);
+            Assert.IsNotNull(form.Fields);
+            var name = "Checkbox_Paramedic Subscription_Amount Enclosed";
+            Assert.IsNotNull(form.Fields[name]);
+            Assert.AreEqual(FieldValueType.SelectionMark, form.Fields[name].Value.ValueType);
+            Assert.AreEqual("selected", form.Fields[name].ValueData.Text);
```
[the remark this answers is not in the document]
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:899]: Thank u for pointing this out. I looked and because Text comes from the service we have it lowercase. I made a modification in `FieldValue_internal` so that we parse text to a `FormSelectionMarkState` and then `ToString` so we are consistent

File: sdk/formrecognizer/Azure.AI.FormRecognizer/README.md [hunk 6]
```diff
@@ -101,7 +101,7 @@ var client = new FormRecognizerClient(new Uri(endpoint), new DefaultAzureCredent
 `FormRecognizerClient` provides operations for:
 
  - Recognizing form fields and content, using custom models trained to recognize your custom forms.  These values are returned in a collection of `RecognizedForm` objects. See example [Recognize Custom Forms](#recognize-custom-forms).
- - Recognizing form content, including tables, lines and words, without the need to train a model.  Form content is returned in a collection of `FormPage` objects. See example [Recognize Content](#recognize-content).
+ - Recognizing form content, including tables, lines, words, and selection marks like radio buttons and check boxes without the need to train a model.  Form content is returned in a collection of `FormPage` objects. See example [Recognize Content](#recognize-content).
```
Comment (kristapratico) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/README.md:104]: I wonder if we should say something about training custom models with labels to recognize selection marks as well

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormSelectionMarkState.cs [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+    [CodeGenModel("SelectionMarkState")]
+    public enum FormSelectionMarkState
+    {
+        /// <summary>
+        /// Value is selected.
+        /// </summary>
+        Selected,
+
+        /// <summary>
+        /// Values is unselected.
```
Comment (kristapratico) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FormSelectionMarkState.cs:20]:
```suggestion
        /// Value is unselected.
```

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FieldValue.cs [hunk 8]
```diff
@@ -349,5 +366,30 @@ public IReadOnlyList<FormField> AsList()
...
+            {
+                return ValueSelectionMark;
+            }
+
+            if (!_fieldValue.ValueSelectionMark.HasValue)
+            {
+                throw new InvalidOperationException($"Field value is null.");
+            }
+
+            return _fieldValue.ValueSelectionMark.Value;
```
Comment (kristapratico) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/src/FieldValue.cs:392]: I might be commenting in the wrong place so correct me if I missed it. I remember that the service team said that it was a bug that valueSelectionMark.value was returning null and that it wouldn't be fixed until 2.1-preview.3. I'm thinking that in the SDK we should return the value found in `text` here in the meantime. What do you think?

File: sdk/formrecognizer/Azure.AI.FormRecognizer/README.md (hunk 6 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/README.md:104]: I thought about it but looking at `Recognizing form fields and content, using custom models trained to recognize your custom forms.` makes me think that it already includes it. Although `selectionmarks` is a new concept, the idea of recognizing fields and content still applies.

File: sdk/formrecognizer/Azure.AI.FormRecognizer/src/FieldValue.cs (hunk 8 above)
Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FieldValue.cs:392]: hmmm we could. I was treating it as we did with the PhoneNumber fieldValue where we return what the service returns.
I am ok making the change and putting a TODO to delete once we confirm the service fixed it

Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/src/FieldValue.cs:392]: Code updated to reflect that

**Review by kristapratico — APPROVED** [review]

File: sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs [hunk 9]
```diff
@@ -812,6 +863,43 @@ public async Task StartRecognizeCustomFormsWithLabels(bool useStream, bool inclu
             Assert.AreEqual("948284", form.Fields[name].ValueData.Text);
         }
 
+        [Test]
+        [TestCase(true)]
+        [TestCase(false)]
+        [Ignore("Need to get an approved file to test")]
```
Comment (samvaity) [inline sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:869]: Are we following up on this?
Is this test still ignored?

Comment (maririos) [reply sdk/formrecognizer/Azure.AI.FormRecognizer/tests/FormRecognizerClient/FormRecognizerClientLiveTests.cs:869]: Yes. I keep pinging service team about it.
And yes, test is ignored
